pkg/channel/attributes/knative_error.go

```diff
@@ -0,0 +1,74 @@
+/*
+Copyright 2021 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package attributes
+
+import (
+	"github.com/cloudevents/sdk-go/v2/binding"
+	"go.uber.org/zap"
+)
+
+const (
+	KnativeErrorCodeExtensionKey       = "knativeerrorcode"
+	KnativeErrorDataExtensionKey       = "knativeerrordata"
+	KnativeErrorDataExtensionMaxLength = 1024
+)
+
+// KnativeErrorCodeTransformer returns a CloudEvent TransformerFunc which sets the specified error code extension.
+func KnativeErrorCodeTransformer(logger *zap.Logger, code int, ignoreErrors bool) binding.TransformerFunc {
+	return func(reader binding.MessageMetadataReader, writer binding.MessageMetadataWriter) error {
+		err := writer.SetExtension(KnativeErrorCodeExtensionKey, code)
+		if err != nil {
+			if logger != nil {
+				logger.Error("Failed to set knative error code extension",
+					zap.String("ExtensionKey", KnativeErrorCodeExtensionKey),
+					zap.Int("ExtensionValue", code),
+					zap.Bool("IgnoreErrors", ignoreErrors),
+					zap.Error(err))
+			}
+			if !ignoreErrors {
+				return err
+			}
+		}
+		return nil
```

Comment:
Why that? Can you just use the built-in transformers?

Reply:
I guess it was for parity with the `KnativeErrorDataTransformer()` implementation (same signatures, etc).  The difference with this and just returning `setExtension()` is that errors will stop the transmission of the events to the dead-letter-sink which I was trying to ensure wouldn't happen (always want them to go even if there is a failure in adding this new error-meta-data).

I'm open to removing the logging in both implementations, and letting errors just fail the transmission, if that's what we want here?  But if we do want the logic in `KnativeErrorDataTransformer()` then I'd vote to keep this implementation aligned for consistency - thoughts?

Reply:
After a quick offline chat... It seems the logging and errorflag aren't really necessary and can be removed in the name of simplicity.  Also going to combine the two extension wrapper functions into one as requested - thanks again!